lms/views/api/moodle/pages.py

```diff
@@ -30,30 +35,73 @@ def list_pages(self):
 
     @view_config(request_method="GET", route_name="moodle_api.pages.via_url")
     def via_url(self):
+        course_copy_plugin = self.request.product.plugin.course_copy
+        current_course_id = self.request.lti_user.lti.course_id
+
         current_course = self.request.find_service(name="course").get_by_context_id(
-            self.request.lti_user.lti.course_id, raise_on_missing=True
+            current_course_id, raise_on_missing=True
         )
         assignment = self.request.find_service(name="assignment").get_assignment(
             self.request.lti_user.application_instance.tool_consumer_instance_guid,
             self.request.lti_user.lti.assignment_id,
         )
         document_url = assignment.document_url
-        _, document_page_id = self._parse_document_url(document_url)
+        document_course_id, document_page_id = self._parse_document_url(document_url)
+        effective_page_id = None
```

Comment:
Very similar to the files version.

Added a note about generalizing this to the board.

Reply:
I would also suggest extracting it into a private method as I did for files. I think it will help generalizing both of them later if we want.